docs/sources/flow/reference/components/loki.process.md

```diff
@@ -530,6 +524,182 @@ Then, the second stage adds `user="John Doe"` to the label set of the log
 entry, and the final output stage changes the log line from the original
 JSON to `hello, world!`.
 
+### match block
+
+The `match` inner block configures a filtering stage that can conditionally
+either apply a nested set of processing stages or drop an entry when a log
+entry matches a configurable LogQL stream selector and filter expressions.
+
+The following arguments are supported:
+
+Name            | Type      | Description                                                         | Default | Required
+--------------- | --------- | ------------------------------------------------------------------- | ------- | --------
+`selector`      | `string`  | The LogQL stream selector and filter expressions to use.            |         | yes
+`pipeline_name` | `string`  | A custom name to use for the nested pipeline.                       | `""`    | no
+`action`        | `string`  | The action to take when the selector matches the log line. Supported values are "keep" and "drop" | `"keep"` | no
+`drop_counter_reason` | `string` | A custom reason to report for dropped lines.                   | `"match_stage"` | no 
```

Comment:
I assume there is a limit on length here? Should we define it.

Reply:
You mean a limit on the length of the custom reason to report? Not sure about that, should be the maximum label length allowed by Loki, I'l look into it.